flowermd/library/systems.py

```diff
@@ -1,3 +1,37 @@
 """Examples for the Systems class."""
 
 # This is a placeholder for any class that inherits from base.system
+import numpy
+from scipy.spatial.distance import pdist
+
+
+class SingleChainSystem(System):
+    """Builds a box around a single chain.
+
+    Calculates the maximum distance of the chain using scipy.spatial.distance.pdist().
+
+    Parameters
+    ----------
+    See System class.
+
+    """
+
+	def __init__(self, molecules, base_units=dict()):
+		super(SingleChainSystem, self).__init__(
+			molecules=molecules,
+			base_units=base_units
+           	 )
+
+        def _build_system(self):
+            chain = self.all_molecules[0]
+            children_pos_array = np.zeros((len(chain.children),3))
+            for i in range(len(chain.children)):
+                children_pos_array[i] = chain.children[i].pos
+            eucl_dist = pdist(children_pos_array)
+            chain_length = np.max(eucl_dist)
+            box = mb.Box(lengths=np.array([chain_length] * 3) * 1.05)
```

Comment:
Let's add a parameter to the `__init__` function for this class that sets this buffer size instead of hard-coding in 1.05.

As a reminder, accessing variables and parameters outside the `__init__` function (for example, using a parameter inside of `_build()`) requires setting a class variable first `self._variable = parameter`